apps/prairielearn/src/ee/lib/aiQuestionGeneration.ts

```diff
@@ -87,14 +139,98 @@ Keep in mind you are not just generating an example; you are generating an actua
       user: openAiUserFromAuthn(authnUserId),
     });
 
-    const completionText = completion.choices[0].message.content;
+    extractFromCompletion(completion, job);
+    job.data['prompt'] = prompt;
+    job.data['generation'] = completion.choices[0].message.content;
+    job.data['context'] = context;
+  });
 
-    job.info(`completion is ${completionText}`);
+  return { jobSequenceId: serverJob.jobSequenceId, completionPromise: jobPromise };
+}
 
-    job.info(`used ${completion?.usage?.total_tokens} OpenAI tokens to generate response.`);
+/**
+ * Revises a question using the LLM based on user input.
+ *
+ * @param client The OpenAI client to use.
+ * @param courseId The ID of the current course.
+ * @param authnUserId The authenticated user's ID.
+ * @param originalPrompt The prompt creating the original generation.
+ * @param revisionPrompt A prompt with user instructions on how to revise the question.
+ * @param originalHTML The question.html file to revise.
+ * @param originalPython The server.py file to revise.
+ * @returns A server job ID for the generation task and a promise to return the associated saved data on completion.
+ */
+export async function regenerateQuestion(
+  client: OpenAI,
+  courseId: string | undefined,
+  authnUserId: string,
+  originalPrompt: string,
+  revisionPrompt: string,
+  originalHTML: string,
+  originalPython: string,
+) {
```

Comment:
Should specify return type.

Reply:
Done in 5af63ea5a9d690527f5feb43c89b6d6ea7c7ba30